File: src/main/kotlin/javacode/MaximumRepeatingString1668.java
package javacode;

public class MaximumRepeatingString1668 {
    public static int maxRepeating(String sequence, String word) {
        int c=0;
        String add=word;
        while(sequence.contains(word)==true)
        {
            c+=1;
            word=word+add;
            System.out.println(word);
        }
        return c;
    }
    public static void main(String[] args) {
        System.out.println(maxRepeating("aaabaaaabaaabaaaabaaaabaaaabaaaaba", "aaaba"));
    }
}

File: src/main/kotlin/kotlincode/MaximumRepeatingString1668.kt
package kotlincode

class MaximumRepeatingString1668 {
}

fun main() {
    var sequence = "aaabaaaabaaabaaaabaaaabaaaabaaaaba"
    var word = "aaaba"
    var toBeConcatenated = word
    var counter = 0
    while (sequence.contains(word)) {
        counter++
        word += toBeConcatenated
        println(word)
    }

    println("Result: $counter")

}